src/Parser/Priest/Shadow/Modules/Items/TwinsPainfulTouch.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS';
+import ITEMS from 'common/ITEMS';
+
+import Enemies from 'Parser/Core/Modules/Enemies';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Analyzer from 'Parser/Core/Analyzer';
+
+import Voidform from '../Spells/Voidform';
+
+
+const LATENCY_BUFFER = 300;
+
+function formatNumber(amount, decimals=1){
+  return Math.round(amount * 10 * decimals) / (10 * decimals);
+}
+
+class TwinsPainfulTouch extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+    voidform: Voidform,
+  };
+
+  totalApplied = 0;
+  mindflayTimestamps = [];
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasFinger(ITEMS.THE_TWINS_PAINFUL_TOUCH.id);
+  }
+
+  on_byPlayer_applydebuff(event){
+    const spellID = event.ability.guid;
+    if(spellID === SPELLS.MIND_FLAY.id && this.combatants.selected.hasBuff(SPELLS.THE_TWINS_PAINFUL_TOUCH.id)){
+      this.mindflayTimestamps.push(event.timestamp);
```

Comment:
Do you really need to track an array of this? How about just tracking the last timestamp?

Reply:
Good call, it's a leftover from earlier. Will get it fixed.